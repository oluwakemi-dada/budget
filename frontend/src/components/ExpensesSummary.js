import React from 'react';
import { useSelector } from 'react-redux';
import numeral from 'numeral';
import visibleExpenses from '../selectors/expenses';
import visibleExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = () => {
  // SELECTOR
  const expenses = useSelector((state) => state.expenses);
  const filters = useSelector((state) => state.filters);

  const visibleExp = visibleExpenses(expenses, filters);

  // VALUES
  const expenseCount = visibleExp.length;
  const expensesTotal = visibleExpensesTotal(visibleExp);

  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const expensesTotalNumeral = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div>
      <h1>
        Viewing {expenseCount} {expenseWord} totaling {expensesTotalNumeral}
      </h1>
    </div>
  );
};

export default ExpensesSummary;
