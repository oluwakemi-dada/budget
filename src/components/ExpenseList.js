import React from 'react';
import { useSelector } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import visibleExpenses from '../selectors/expenses';

const ExpenseList = () => {
  // SELECTOR
  const expenses = useSelector((state) => state.expenses);
  const filters = useSelector((state) => state.filters)

  const visibleExp = visibleExpenses(expenses, filters)

  return (
    <div>
      {expenses.length === 0 ? (
        <p>No expenses</p>
      ) : (
        visibleExp.map((expense) => (
          <ExpenseListItem key={expense.id} {...expense} />
        ))
      )}
    </div>
  );
};

export default ExpenseList;
