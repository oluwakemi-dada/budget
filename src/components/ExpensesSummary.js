import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import visibleExpenses from '../selectors/expenses';
import visibleExpensesTotal from '../selectors/expenses-total';

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => {
  const expenseWord = expenseCount === 1 ? 'expense' : 'expenses';
  const expensesTotalNumeral = numeral(expensesTotal / 100).format('$0,0.00');
  return (
    <div>
      <h1>
        Viewing {expenseCount} {expenseWord} totaling {expensesTotalNumeral}
      </h1>
    </div>
  );
};

const mapStateToProps = (state) => {
  const getVisibleExpenses = visibleExpenses(state.expenses, state.filters);
  return {
    expenseCount: getVisibleExpenses.length,
    expensesTotal: visibleExpensesTotal(getVisibleExpenses),
  };
};

export default connect(mapStateToProps)(ExpensesSummary);
