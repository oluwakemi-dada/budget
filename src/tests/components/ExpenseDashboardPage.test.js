import React from 'react';
import { shallow } from 'enzyme';
import ExpenseList from '../../components/ExpenseList';
import ExpenseListFilter from '../../components/ExpenseListFilters';
import ExpensesSummary from '../../components/ExpensesSummary';

test('should render ExpenseDashboardPage correctly', () => {
  const wrapper = shallow(
    <div>
      <ExpensesSummary />
      <ExpenseListFilter />
      <ExpenseList />
    </div>
  );
  expect(wrapper).toMatchSnapshot();
});
