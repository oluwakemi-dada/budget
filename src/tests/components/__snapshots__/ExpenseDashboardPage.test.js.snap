// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseDashboardPage correctly 1`] = `
<div>
  <ExpensesSummary />
  <ExpenseListFilter />
  <ExpenseList />
</div>
`;
